### ISSUE TITLE
"stack size does not match block type" on valid wasm

### ISSUE THREAD

**Issue by kripken** [wasmerio/wasmer#815]
```shell
$ wasmer run a1.wasm
execute_wasm: "Can\'t compile module: ValidationError { msg: \"stack size does not match block type\" }"
```

The wasm file is valid, though.

Tested on wasmer 0.6.0, x86_64.

This was found by the binaryen fuzzer. Apologies it isn't smaller - the problem seems to be something to do with unreachable code, because when the reducer tries to shrink the testcase it fails.

[a1.wasm.zip](https://github.com/wasmerio/wasmer/files/3628632/a1.wasm.zip)

**Comment by nlewycky** [comment]
This is a bug in the version of wasmparser.rs we're using which appears to be fixed at head.

**Comment by syrusakbary** [comment]
Note: this appears to be fixed in wasmparser.rs, but the issue is still happening in latest wasmer (`0.11.0`).

**Comment by syrusakbary** [comment]
I've checked again, and this issue is now fixed in `master` (thanks to #1777 ).

The next release of Wasmer will be able to run the a1.wasm file provided :)
